outlines/fsm/fsm.py

```diff
@@ -121,9 +156,22 @@ def create_states_mapping(
         self.states_to_token_maps, self.empty_token_ids = create_states_mapping(
             regex_string, tuple(sorted(tokenizer.vocabulary.items()))
         )
-        self.vocabulary = tokenizer.vocabulary.values()
+        self.vocabulary = tokenizer.vocabulary
         self.eos_token_id = tokenizer.eos_token_id
 
+    def align_prompt_tokens(
```

Comment:
Is there any way we could avoid modifying the `outlines.fsm.fsm` file and implement all the logic in `outlines.generate.api`? These feel like separate concerns.

Reply:
My issue was that this token alignment does not apply to CFGFSM as it does not have a `states_to_token_maps`